packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -190,3 +219,98 @@ export function getNetworkName(chainId: number) {
       return 'Unknown'
   }
 }
+
+export function getNetworkLogo(chainId: number) {
+  switch (chainId) {
+    // L1 networks
+    case ChainId.Mainnet:
+    case ChainId.Rinkeby:
+    case ChainId.Goerli:
+      return EthereumLogo
+
+    // L2 networks
+    case ChainId.ArbitrumOne:
+    case ChainId.ArbitrumRinkeby:
+    case ChainId.ArbitrumGoerli:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumNova:
+      return ArbitrumNovaLogo
+
+    default:
+      return EthereumLogo
+  }
+}
+
+export type SwitchChainProps = {
+  chainId: number
+  provider: ExtendedWeb3Provider
+  onSuccess?: () => void
+  onError?: (err?: Error) => void
+  onSwitchChainNotSupported?: () => void
+}
+
+const isSwitchChainSupported = (provider: ExtendedWeb3Provider) => {
+  const { provider: innerProvider } = provider
+  // @ts-ignore : `isImToken` is not exported by default in metamask
+  return innerProvider?.isMetaMask || innerProvider?.isImToken
+}
```

Comment:
Our typing seems to be wrong here, as what we call the `innerProvider` should be typed as `ExtendedWeb3Provider`.